test/e2e/policies.test-suite/balloons/n4c16/test10-health-checking/code.var.sh

```diff
@@ -1,17 +1,29 @@
 # Test that
 # Syntatically incorrect configuration file is resulting in the nri-resmgr
 # pod readiness probe failure.
+dummyData="foo: bar"
 vm-put-file $(instantiate nri-resmgr-configmap.yaml) nri-resmgr-configmap.yaml
 
 terminate nri-resmgr
-launch nri-resmgr
+nri_resmgr_config=fallback launch nri-resmgr
 vm-command "kubectl apply -f nri-resmgr-configmap.yaml"
 
-# Give enough time for the Kubernetes to consider the container unhealthy
+# Give enough time for the Kubernetes to consider the container not ready.
 sleep 10
 
 # Check if nri-resmgr readiness probe failed as expected
 if ! vm-command "kubectl get event -n kube-system --field-selector involvedObject.name=$POD | grep -q 'Readiness probe failed'" 2>&1; then
-    error "Expected liveness probe to fail, but got it succeeded"
+    error "Expected readiness probe to fail, but got it succeeded"
 fi
-echo "nri-resmgr health-check failed as expected"
+echo "nri-resmgr readiness probe failed as expected"
+
+# Fix the incorrect data in the configuration and check if nri-rm becomes ready.
+dummyData=""
+vm-put-file $(instantiate nri-resmgr-configmap.yaml) nri-resmgr-configmap.yaml
+
+sleep 5
+
+if vm-command "kubectl get event -n kube-system --field-selector involvedObject.name=$POD | grep -q 'Readiness probe failed'" 2>&1; then
```

Comment:
I think instead of the `sleep 5; kubectl get event...`-combo it would be better to do something like

`kubectl -n kube-system wait --for=condition=Ready=false pods/$POD --timeout=60s`

That would allow both a more relaxed timeout, yet not waste any time once the readiness transition has happened.

Reply:
Actually I stopped using kubectl event in https://github.com/jukkar/nri-resmgr/pull/66/commits/3ca4397c11f3ca0d69a4c135dbae5e2e19eabc0a but I missed to update the same for balloons. 

But for the timeout, I think it is better to use the kubectl flag instead.

Reply:
> Actually I stopped using kubectl event in [3ca4397](https://github.com/jukkar/nri-resmgr/commit/3ca4397c11f3ca0d69a4c135dbae5e2e19eabc0a) but I missed to update the same for balloons.
> 
> But for the timeout, I think it is better to use the kubectl flag instead.

But would that work as desired when used together with `kubect get {event|pod}` ? I think not. Those would return the current event or pod state without waiting for anything else than an response from the API server which will be pretty much immediate. That is very different from `kubectl wait` where you tell what condition you want kubectl to *wait for* and the timeout indicates the period after which `kubectl` should consider the condition/transition to have failed to happen.

If your idea was to optimize (IOW shorten) how fast the tests can succeed, then wouldn't it be better to simply increase the frequency of the readiness probe enough and use `kubectl wait` ?

Reply:
> vm-command "kubectl -n kube-system wait --for=condition=Ready=true pods/$POD --timeout=60s"

instead of this, we have `vm-wait-pod-regexp()` that does exacty that

Reply:
@klihub sorry you were right. I assumed that I could use --timeout with kubect get pod, which was a wrong assumption. 
What you have proposed is much simpler and cleaner. I have updated the commit now accordingly. Thanks for the suggestion.

Reply:
> > vm-command "kubectl -n kube-system wait --for=condition=Ready=true pods/$POD --timeout=60s"
> 
> instead of this, we have `vm-wait-pod-regexp()` that does exacty that

I think that function has some limitations for our use case, for example `kubectl wait --timeout=${wait_t}s --for=condition=${wait}` line doesn't have true/false option, it only looks for condition but not the value. I can extend it and I just tried it but it is getting out of scope of this PR. TBH, I'm not super happy about using that function in here, because it is a wrapper which complicates a simple kubectl command. Perhaps it is useful in other scenarios but not sure here ?

Reply:
Althought I already +1, I think we should change this a bit as suggested by Krisztian,
to use the `vm-wait-pod-regexp()` function

Reply:
I'm fine/LGTM with this either way...  `vm-wait-pod-regexp()` or vanilla `kubectl wait --for`. No strong opinions here. You guys can decide.

Reply:
I didn't notice that we can pass anything into the wait. Adding wait=Ready=true/false worked. Thanks

Reply:
sorry for delay @klihub @jukkar . I wasted my time on by adding condition value incorrectly and searching for the issue in a wrong place. I have fixed it now. PTAL.